core/solver/ir.cpp

```diff
@@ -160,21 +160,31 @@ std::unique_ptr<LinOp> Ir<ValueType>::conj_transpose() const
 
 template <typename ValueType>
 void Ir<ValueType>::apply_impl(const LinOp* b, LinOp* x) const
+{
+    this->apply_with_initial_guess(b, x, this->get_default_initial_guess());
+}
+
+
+template <typename ValueType>
+void Ir<ValueType>::apply_with_initial_guess(const LinOp* b, LinOp* x,
```

Comment:
This matches the interface of `apply`, but doesn't validate the parameters or call `make_temporary_clone`